Core.DomainServices/Repositories/Notification/UserNotificationRepository.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Core.DomainModel.Notification;
+using Core.DomainModel.Result;
+using Core.DomainModel.Shared;
+using Core.DomainServices.Extensions;
+using Infrastructure.Services.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Core.DomainServices.Repositories.Notification
+{
+    public class UserNotificationRepository : IUserNotificationRepository
+    {
+        private readonly IGenericRepository<UserNotification> _repository;
+
+        public UserNotificationRepository(IGenericRepository<UserNotification> repository)
+        {
+            _repository = repository;
+        }
+
+        public UserNotification Add(UserNotification newNotification)
+        {
+            var notification = _repository.Insert(newNotification);
+            _repository.Save();
+            return notification;
+        }
+
+        public Maybe<OperationError> DeleteById(int id)
+        {
+            var notification = _repository.GetByKey(id);
+            if (notification != null)
+            {
+                _repository.Delete(notification);
+                _repository.Save();
+                return Maybe<OperationError>.None;
+            }
+
+            return new OperationError(OperationFailure.UnknownError);
+        }
+
+        public Maybe<UserNotification> GetById(int id)
+        {
+            return _repository.GetByKey(id);
+        }
+
+        public IEnumerable<UserNotification> GetByRelatedEntityIdAndType(int relatedEntityId, RelatedEntityType relatedEntityType)
+        {
+            switch (relatedEntityType)
+            {
+                case RelatedEntityType.itContract:
+                    return _repository.Get(x => x.Itcontract_Id == relatedEntityId);
```

Comment:
Brug _respository.AsQueryable() i stedet for den der--- den terminerer query til ienumerable